### PR TITLE
DOC: Add note for copy keywords for Copy-on-Write

### PR DESCRIPTION
**Pull request by phofl**
- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

Starting with one method to figure out wording. Will add for the remaining ones after this one

### CODE REVIEW

File: pandas/core/generic.py [hunk 1]
```diff
@@ -6428,6 +6428,15 @@ def astype(
             Return a copy when ``copy=True`` (be very careful setting
             ``copy=False`` as changes to values then may propagate to other
             pandas objects).
+
+            .. note::
```
Comment (lithomas1) [inline pandas/core/generic.py:6432]: Can you make this a shared docstring?

File: pandas/core/generic.py [hunk 2]
```diff
@@ -6428,6 +6428,15 @@ def astype(
...
+                The copy-keyword will change behavior in pandas 3.0.
+                `Copy-on-Write
+                <https://pandas.pydata.org/docs/dev/user_guide/copy_on_write.html>`__
+                will be enabled by default, which means that all methods with a
+                copy-keyword will use a lazy copy mechanism and ignore the copy
```
Comment (jorisvandenbossche) [inline pandas/core/generic.py:6437]:
```suggestion
                `copy` keyword will use a lazy copy mechanism and ignore the `copy`
```

File: pandas/core/generic.py [hunk 3]
```diff
@@ -6428,6 +6428,15 @@ def astype(
...
+                keyword. The copy-keyword will be removed in a future version of
```
Comment (jorisvandenbossche) [inline pandas/core/generic.py:6438]:
```suggestion
                keyword. The `copy` keyword will be removed in a future version of
```

File: pandas/core/generic.py [hunk 4]
```diff
@@ -6428,6 +6428,15 @@ def astype(
             Return a copy when ``copy=True`` (be very careful setting
             ``copy=False`` as changes to values then may propagate to other
             pandas objects).
+
+            .. note::
+                The copy-keyword will change behavior in pandas 3.0.
```
Comment (jorisvandenbossche) [inline pandas/core/generic.py:6433]:
```suggestion
                The `copy` keyword will change behavior in pandas 3.0.
```

**Review by jorisvandenbossche — COMMENTED** [review]
Looks good to me.

I might want to say something like: "in the future this will no longer copy by default, so you don't need to specify this keyword", but the problem is in being both succinct and accurate, because of course that isn't fully true, we will copy later on. Which is what your "lazy copy" covers, but then I don't know how many people will fully understand that. 

Do we want to also make use of this note to mention "you can already get that future behaviour by enabling CoW, see ...` ?

File: pandas/core/generic.py (hunk 1 above)
Comment (jorisvandenbossche) [reply pandas/core/generic.py:6432]: Personally I find shared docstrings quite annoying (and untransparant, difficult to understand), so while it's nice to deduplicate and only have 1 place to update this note, I am also fine with simply copy-pasting this note multiple times once we have finalized its wording.

(certainly for methods that right now don't yet make use of shared docstrings)

Comment (phofl) [reply pandas/core/generic.py:6432]: Yeah I don't like them either

**Comment by phofl** [comment]
going ahead with the others now

**Pull request merged**